src/openrct2/Date.cpp

```diff
@@ -0,0 +1,129 @@
+#pragma region Copyright (c) 2014-2017 OpenRCT2 Developers
+/*****************************************************************************
+* OpenRCT2, an open source clone of Roller Coaster Tycoon 2.
+*
+* OpenRCT2 is the work of many authors, a full list can be found in contributors.md
+* For more information, visit https://github.com/OpenRCT2/OpenRCT2
+*
+* OpenRCT2 is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* A full copy of the GNU General Public License can be found in licence.txt
+*****************************************************************************/
+#pragma endregion
+
+#include "core/Guard.hpp"
+#include "core/Math.hpp"
+#include "Date.h"
+
+using namespace OpenRCT2;
+
+constexpr sint32 MONTH_TICKS_INCREMENT = 4;
+constexpr sint32 MASK_WEEK_TICKS = 0x3FFF;
+constexpr sint32 MASK_FORTNIGHT_TICKS = 0x7FFF;
+constexpr sint32 MASK_MONTH_TICKS = 0xFFFF;
+
+Date::Date()
+{
+}
+
+Date::Date(uint32 monthsElapsed, uint16 monthTicks)
+    : _monthsElapsed(monthsElapsed),
+      _monthTicks(monthTicks)
+{
+}
+
+Date Date::FromYMD(sint32 year, sint32 month, sint32 day)
```

Comment:
The constructor uses uint32 and uint16 because it is just setting the underlying fields. This is a helper and therefore uses the most common integral data type: `int`.